lib/Sema/TypeCheckAccess.cpp

```diff
@@ -659,55 +685,54 @@ class AccessControlChecker : public AccessControlCheckerBase,
     if (!anyVar)
       return;
 
-    checkTypeAccess(TP->hasType() ? TP->getType() : Type(),
-                    TP->getTypeRepr(), anyVar, /*mayBeInferred*/true,
-                    [&](AccessScope typeAccessScope,
-                        const TypeRepr *complainRepr,
-                        DowngradeToWarning downgradeToWarning,
-                        ImportAccessLevel importLimit) {
-      auto typeAccess = typeAccessScope.accessLevelForDiagnostics();
-      bool isExplicit = anyVar->getAttrs().hasAttribute<AccessControlAttr>() ||
-                        isa<ProtocolDecl>(anyVar->getDeclContext());
-      auto diagID = diag::pattern_type_access;
-      if (downgradeToWarning == DowngradeToWarning::Yes)
-        diagID = diag::pattern_type_access_warn;
-      auto anyVarAccess =
-          isExplicit ? anyVar->getFormalAccess()
-                     : typeAccessScope.requiredAccessForDiagnostics();
-      auto &DE = anyVar->getASTContext().Diags;
-      auto diag = DE.diagnose(
-          TP->getLoc(), diagID, anyVar->isLet(), isTypeContext, isExplicit,
-          anyVarAccess, isa<FileUnit>(anyVar->getDeclContext()), typeAccess);
-      highlightOffendingType(diag, complainRepr);
-      noteLimitingImport(anyVar, importLimit, complainRepr);
-    });
+    checkTypeAccess(
+        TP->hasType() ? TP->getType() : Type(), TP->getTypeRepr(), anyVar,
+        /*mayBeInferred*/ true,
+        [&](AccessScope typeAccessScope, const TypeRepr *complainRepr,
```

Comment:
Do we really need to change the signature to pass `complainDecl` through? I may be missing the counter-example, but it looks to me like the decl could just be captured by the lambda to avoid all of the churn of taking the decl as an argument to `checkTypeAccess` and the callback.

Reply:
Updated to move the logic to the checkNamedPattern callback instead, which avoids all the other unrelated callback changes.